dao/src/main/resources/db/migration/V65__removeUserInfoFromRepositoryUrl.sql

```diff
@@ -0,0 +1,31 @@
+-- Remove user information (e.g. username:password@ or username@) from the repository's URL.
+-- Delete entry if there two repositories with the same URL after removing the user information.
+
+WITH clean_entries AS (
+    SELECT
+        id,
+        product_id,
+        url,
+        regexp_replace(
+                url,
+                '^(https?://)([^/@:]+(:[^/@]*)?@)?',
+                '\1'
+        ) AS clean_url,
+        ROW_NUMBER() OVER (
+            PARTITION BY
+                regexp_replace(url, '^(https?://)([^/@:]+(:[^/@]*)?@)?', '\1'),
+                product_id
+            ORDER BY id
+        ) AS rn
+    FROM repositories)
+DELETE
```

Comment:
Will this DELETE work if the repository is referenced from ORT runs? This would violate a foreign key relation, wouldn't it?

Reply:
Good point. As discussed, I will change my SQL query so that no repositories will be deleted. Instead, it will replace the user information with something like 'xxxx' (from https://username@github.com/org/repo.git to https://xxxx@github.com/org/repo.git).